lib/src/rules/prefer_is_not_empty.dart

```diff
@@ -63,48 +57,39 @@ class _Visitor extends SimpleAstVisitor<void> {
   _Visitor(this.rule);
 
   @override
-  void visitSimpleIdentifier(SimpleIdentifier node) {
-    late AstNode isEmptyAccess;
-    SimpleIdentifier? isEmptyIdentifier;
-
-    var parent = node.parent;
-    if (parent is PropertyAccess) {
-      isEmptyIdentifier = parent.propertyName;
-      isEmptyAccess = parent;
-    } else if (parent is PrefixedIdentifier) {
-      isEmptyIdentifier = parent.identifier;
-      isEmptyAccess = parent;
+  void visitPrefixExpression(PrefixExpression node) {
+    // Should be prefixed w/ a "!".
+    var prefix = node.operator;
+    if (prefix.type != TokenType.BANG) {
+      return;
     }
 
+    var expression = node.operand.unParenthesized;
+
+    // Should be a property access or prefixed identifier.
+    SimpleIdentifier? isEmptyIdentifier;
```

Comment:
Not important, but you could probably make this non-nullable (and remove the check on line 76) by rewriting slightly as
```
if (expression is Prop...) {
  ...
} else if (...) {
  ...
} else {
  return;
}
```